src/index.js

```diff
@@ -1,56 +1,69 @@
-require('./structures/message')
+require('./structures/message');
 
-const { Client, MessageAttachment, APIMessage, Intents } = require('discord.js')
-const path = require('path')
+const {
+  Client,
+  MessageAttachment,
+  MessagePayload,
+  Intents,
+} = require('discord.js');
+const path = require('path');
 const pool = require('workerpool').pool(path.join(__dirname, './worker.js'), {
   workerType: 'process',
-})
+});
+const { codeBlock } = require('@discordjs/builders');
+const intents =
+  Intents.FLAGS.GUILDS |
+  Intents.FLAGS.GUILD_MESSAGES |
+  Intents.FLAGS.GUILD_MESSAGE_REACTIONS;
 
 const client = new Client({
+  intents,
   ws: {
-    intents: Intents.NON_PRIVILEGED,
+    intents: intents,
   },
   presence: {
     activity: {
       name: 'JavaScript',
       type: 'PLAYING',
     },
   },
-})
-
-const codeBlockRegex = /^`{3}(?<lang>[a-z]+)\n(?<code>[\s\S]+)\n`{3}$/mu
-const languages = ['js', 'javascript']
+});
 
+const BlockRegex = /^`{3}(?<lang>[a-z]+)\n(?<code>[\s\S]+)\n`{3}$/mu;
+const languages = ['js', 'javascript'];
 const toMessageOptions = content => {
-  if (content.length <= 2000)
-    return APIMessage.transformOptions(content, { code: 'js' })
-  else
-    return APIMessage.transformOptions(
-      '実行結果が長すぎるのでテキストファイルに出力しました。',
-      new MessageAttachment(Buffer.from(content), 'result.txt')
-    )
-}
-
-client.once('ready', () => console.log('Ready'))
-
-client.on('message', message => {
-  if (message.author.bot || message.system) return
-  if (!message.content.toLowerCase().startsWith('>runjs')) return
-  if (!codeBlockRegex.test(message.content))
-    return message.reply('コードを送信してください。').catch(console.error)
-
-  const codeBlock = message.content.match(codeBlockRegex)?.groups ?? {}
-
-  if (!languages.includes(codeBlock.lang))
+  if (content.length <= 2000) return codeBlock('js', content);
+  else {
+    const file = new MessageAttachment().setFile(
+      Buffer.from(content),
+      'result.js'
+    );
```

Comment:
Discord.jsの[ソースコード](https://github.com/discordjs/discord.js/blob/stable/src/structures/MessageAttachment.js#L8)を見ると、これでも確かに動作するのですが[ドキュメント](https://discord.js.org/#/docs/discord.js/stable/class/MessageAttachment)にはコンストラクタに渡す最初の値がOptionalではないというのと、なぜファイル名を`result.js`にされたのか分かりませんがJavaScriptのコードを出力するものではないので`result.txt`で問題ないと考えています。

以上の２つで、ここは下記のコードで大丈夫です。

```suggestion
    const file = new MessageAttachment(Buffer.from(content), 'result.txt')
```